contracts/axelar-gas-service/tests/test.rs

```diff
@@ -181,6 +181,91 @@ fn pay_gas_for_contract_call() {
     );
 }
 
+#[test]
+fn fail_add_gas_zero_gas_amount() {
+    let (env, contract_id, _gas_collector, client) = setup_env();
+
+    let asset = env.register_stellar_asset_contract_v2(Address::generate(&env));
+    let sender: Address = Address::generate(&env);
+    let message_id = String::from_str(
+        &env,
+        &format!(
+            "{}-{}",
+            "0xfded3f55dec47250a52a8c0bb7038e72fa6ffaae33562f77cd2b629ef7fd424d", 0
+        ),
+    );
+    let refund_address: Address = Address::generate(&env);
+    let gas_amount: i128 = 0;
+    let token = Token {
+        address: asset.address(),
+        amount: gas_amount,
+    };
+
+    let token_client = TokenClient::new(&env, &asset.address());
+    StellarAssetClient::new(&env, &asset.address()).mint(&sender, &gas_amount);
+
+    let expiration_ledger = &env.ledger().sequence() + 200;
+
+    // approve token spend before invoking `add_gas`
+    token_client.approve(&sender, &contract_id, &gas_amount, &expiration_ledger);
+
+    assert_eq!(token_client.allowance(&sender, &contract_id), gas_amount);
+
+    assert_contract_err!(
+        client.try_add_gas(&sender, &message_id, &refund_address, &token,),
+        ContractError::InvalidAmount
+    );
+}
+
+#[test]
+fn add_gas() {
+    let (env, contract_id, _gas_collector, client) = setup_env();
+
+    let asset = env.register_stellar_asset_contract_v2(Address::generate(&env));
+    let sender: Address = Address::generate(&env);
+    let message_id = String::from_str(
+        &env,
+        &format!(
+            "{}-{}",
+            "0xfded3f55dec47250a52a8c0bb7038e72fa6ffaae33562f77cd2b629ef7fd424d", 0
+        ),
+    );
+    let refund_address: Address = Address::generate(&env);
+    let gas_amount: i128 = 1;
+    let token = Token {
+        address: asset.address(),
+        amount: gas_amount,
+    };
+
+    let token_client = TokenClient::new(&env, &asset.address());
+    StellarAssetClient::new(&env, &asset.address()).mint(&sender, &gas_amount);
+
+    let expiration_ledger = &env.ledger().sequence() + 200;
+
+    // approve token spend before invoking `add_gas`
```

Comment:
```suggestion
```
i think it's implied, since it's similar to ERC20

Reply:
removed comment